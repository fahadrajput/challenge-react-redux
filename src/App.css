body, html {
	height: 100%;
}

#root {
	height: 100%;
}

.App {
	margin: 0 3em;
	height: 100%;
	background-color: white;
}

.App-header {
	text-align: center;
}

.App-nav {
	margin-bottom: 2em;
}

.App-nav a {
	text-decoration: none;
	color: #36459b;
	margin: 0 1em;
}

.App-nav a:hover {
	text-decoration: underline;
}

.App-nav a:first-child {
	margin-left: 0;
}

.App-link {
	color: #61dafb;
}


.machine1{
	display: flex;
}

.input-update{
	width: 80%;
	margin-bottom: 20px;
	display: block;
}

.submit{
	display: flex;
	justify-content: flex-end;
	margin-right: 20%;
}

.health{
	width: 80%;
	border: 1px black solid;
	margin-bottom: 20px;
}

.machine-name{
	text-align: center;
	margin-top: 10px;
	margin-bottom: -5px;
}