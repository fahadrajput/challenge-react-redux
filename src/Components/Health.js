/* eslint-disable indent */
import React from 'react';
import 'antd/dist/antd.css';


export default function Health(v) {

    return (
        <div>
            <div style={{ display: 'flex', justifyContent: 'center' }}>
                <div className="progress" style={{ width: '90%', marginBottom: 10, marginTop: 10 }}>
                    <div className="progress-bar" role="progressbar" aria-valuenow="70" aria-valuemin="0" aria-valuemax="100" style={{ width: `${v.health}%`, backgroundColor: v.health <= 50 ? '#5cb85c' : v.health >= 51 & v.health <= 70 ? '#f0ad4e' : '#f0ad4e' }}>
                    </div>
                </div>
            </div>
        </div>
    );
}
