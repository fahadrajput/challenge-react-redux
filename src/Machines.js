import React, { useEffect, useState } from 'react';
import 'antd/dist/antd.css';
import { Table, Progress, Button } from 'antd';
import axios from 'axios';


const columns = [
	{
		title: 'Name',
		dataIndex: 'name'
	},
	{
		title: 'IP Address',
		dataIndex: 'ip_address'
	},
	{
		title: 'Health',
		dataIndex: 'health',
		render: (v) => <div>
			<Progress percent={v} strokeWidth={15} size="small" />
		</div>
	}
];


export default function Machines() {
	const [loading, setLoading] = useState(true)
	const [data, setData] = useState([])

	useEffect(() => {
		axios.get('http://localhost:8080/machines')
			.then((result) => {
				const { data } = result
				setData(data)
				setLoading(false)
			})
	})

	return (
		<div>
			<div>Machines view</div>
			<Table
				columns={columns}
				dataSource={data}
				size="middle"
				pagination={false}
				loading={loading}
				onRow={(record, rowIndex) => {
					return {
						onClick: () => {
							console.log(record.id)
						}
					};
				}}
			/>
		</div>
	);
}
