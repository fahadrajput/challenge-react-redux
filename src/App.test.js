import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import Machines from './Machines'
import Health from './Components/Health'
import { useHistory, useParams } from 'react-router-dom'
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Link
} from 'react-router-dom';
import renderer from 'react-test-renderer';

test('renders /machines link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/machines/i);
  expect(linkElement).toBeInTheDocument();
})


describe('Health', () => {
  test('render Health Progress', () => {
    const component = renderer.create(<Health health={70} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Health', () => {
  test('render Machines list Progress', () => {
    const component1 = renderer.create(<Machines useHistory={useHistory()} />);
    let tree = component1.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
