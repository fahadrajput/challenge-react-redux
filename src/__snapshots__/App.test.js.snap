// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter snapshot renders 1`] = `
<div
  data-testid="table"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": "90%",
        }
      }
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="70"
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#5cb85c",
            "width": "1%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Health rend 1`] = `
<div
  data-testid="table"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": "90%",
        }
      }
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="70"
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#f0ad4e",
            "width": "70%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Health render Health Progress 1`] = `
<div
  data-testid="table"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": "90%",
        }
      }
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="70"
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#f0ad4e",
            "width": "70%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Health render Health Progress 2`] = `
<div
  data-testid="table"
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="progress"
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
          "width": "90%",
        }
      }
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="70"
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "backgroundColor": "#f0ad4e",
            "width": "70%",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Health render Machines list Progress 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      height="272"
      src="https://i.imgur.com/jcvsFKh.png"
      width="800"
    />
  </header>
  <nav
    className="App-nav"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/machines"
      onClick={[Function]}
    >
      Machines
    </a>
  </nav>
</div>
`;
